specs/deposits.md

```diff
@@ -44,41 +69,36 @@ Picking a high identifier minimizes the risk that the identifier will be used be
 transaction type on the L1 chain in the future. We don't pick `0x7F` itself in case it becomes used
 for a variable-length encoding scheme.
 
-### Uses of the Deposit Transaction Type
+### Kinds of Deposited Transactions
 
-Although in practice we define only one new transaction type we can distinguish between two distinct
-situations which occur in the deposit block, based on their positioning.
+Although we define only one new transaction type, we can distinguish between two kinds of deposited
+transactions, based on their positioning in the [deposit block][g-deposit-block]:
 
-1. The first transaction MUST be a [L1 attributes deposit][l1-attributes-deposit], followed by
-2. an array of zero-or-more [L1 transaction deposits][l1-transaction-deposits] submitted to the
-deposit feed contract on L1.
+1. The first transaction MUST be a [L1 attributes deposited transaction][l1-attr-deposit], followed
+   by
+2. an array of zero-or-more [user-deposited transactions][user-deposited] submitted to the deposit
+   feed contract on L1.
 
-The rationale for creating only one new transaction type is to minimize both
-modifications to L1 client software and complexity in general.
+We only define a single new transaction type in order to minimize modifications to L1 client
+software, and complexity in general.
 
-> **TODO** Specify and link to deposit blocks
+### Validation and Authorization of Deposited Transactions
 
-### Validation and Authorization of Deposit Transaction Types
+[authorization]: #validation-and-authorization-of-deposited-transaction
 
-[authorization]: #validation-and-authorization-of-deposit-transaction-types
-
-As noted above, the deposit transaction type does not include a signature for validation. Rather,
-authorization is handled by the [L2 chain Derivation][derivation] process, which when correctly
-processed will only derive transactions with a `from` address attested to by the logs of the [L1
+As noted above, the deposited transaction type does not include a signature for validation. Rather,
+authorization is handled by the [L2 chain derivation][g-derivation] process, which when correctly
+applied will only derive transactions with a `from` address attested to by the logs of the [L1
 deposit feed contract][deposit-feed-contract].
 
-In the event a deposit transaction is included which is not derived by the [execution
-engine][execution-engine] using the correct derivation algorithm, the resulting state transition
-would be invalid.
-
```

Comment:
Removed this part because it was erroneous (it's not about the execution engine but the rollup node) and also because we can't really speak to the consequences at this stage (since we don't submit roots on L1) - what does it mean to be invalid? A bad transaction being included can also not lead to any state transition (i.e. the only bad output is in the block hash, but the state root is correct). Also if you don't follow the prescribed rules, then the result is invalid, which is sort of tautological.

Reply:
The "valid"/"invalid" refers to following the verifier rules I think. The local rollup-node can insert any deposit into their execution-engine, but if it does not originate from the specified derivation, other nodes will regard it as invalid. You're right that while we don't have block propagation or state-root posting it's not applicable (as other nodes cannot consume the invalid data in the first place).